cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -131,13 +159,36 @@ public Future<KafkaStatus> createOrUpdate(Reconciliation reconciliation, Kafka k
                 status.setClusterId(kafkaAssembly.getStatus().getClusterId());
             }
 
+            if (status.getOperatorLastSuccessfulVersion() == null
```

Comment:
Can `status.getOperatorLastSuccessfulVersion() == null` really be false at any point? You seem to set it only at line 191 which is behind this point.